smoke-tests/runner/src/test/java/io/awsobservability/instrumentation/smoketests/runner/SpringBootSmokeTest.java

```diff
@@ -186,6 +201,35 @@ void hello() {
             });
   }
 
+  @Test
+  void defaultSamplingRate() {
+    int numRequests = 100;
+    for (int i = 0; i < numRequests; i++) {
+      appClient.get("/hello").aggregate().join();
+    }
+
+    var exported = getExported();
+    // 5 spans per request (1 CLIENT, 2 SERVER, 2 INTERNAL)
+    assertThat(exported).hasSize(numRequests * 5);
+  }
+
+  @Test
+  void remoteSamplingRate() {
+    // We just want to make sure OTEL_TRACES_SAMPLER=xray is using the XRay sampler. It will fail to
+    // fetch rules and fallback to the default, which will not sample all requests.
+    WebClient client =
+        WebClient.of("http://localhost:" + applicationXraySampler.getMappedPort(8080));
+    int numRequests = 100;
+    for (int i = 0; i < numRequests; i++) {
+      client.get("/hello").aggregate().join();
+    }
+
+    var exported = getExported();
+    // 5 spans per request (1 CLIENT, 2 SERVER, 2 INTERNAL). The default sampler is around 5%, we do
```

Comment:
Is sampling in otel applied to every span individually regardless of Kind or parent? I would expect it to only be applied to the root span like what X-Ray does.

Reply:
Yeah OTel has the ParentBased sampler, which is the default that has that behavior. So 5% is applied to the root which then will either enable or disable the rest of the spans in the trace. I added the word "if sampled" to my comment, does it make it less confusing?

Reply:
Ah yep that clarifies it